libs/ngrx-toolkit/src/lib/mutation/http-mutation.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  HttpClient,
+  HttpContext,
+  HttpEventType,
+  HttpHeaders,
+  HttpParams,
+  HttpProgressEvent,
+  HttpResponse,
+} from '@angular/common/http';
+import { inject, Signal, signal } from '@angular/core';
+import { defer, filter, map, tap } from 'rxjs';
+import { Mutation } from './mutation';
+import { rxMutation, RxMutationOptions } from './rx-mutation';
+
+// The HttpClient defines these types as part of the method signature and
+// not as a named type
+export type HttpMutationRequest = {
+  url: string;
+  method: string;
+  body?: unknown;
+  headers?: HttpHeaders | Record<string, string | string[]>;
+  context?: HttpContext;
+  reportProgress?: boolean;
+  params?:
+    | HttpParams
+    | Record<
+        string,
+        string | number | boolean | ReadonlyArray<string | number | boolean>
+      >;
+  withCredentials?: boolean;
+  credentials?: RequestCredentials;
+  keepalive?: boolean;
+  priority?: RequestPriority;
+  cache?: RequestCache;
+  mode?: RequestMode;
+  redirect?: RequestRedirect;
+  transferCache?:
+    | {
+        includeHeaders?: string[];
+      }
+    | boolean;
+};
+
+export type HttpMutationOptions<Parameter, Result> = Omit<
```

Comment:
Can we please add an optional `parse` here and change the type to the following?

PLEASE DON'T SKIP THE `NoInfer` in `RxMutationOptions`. THAT ONE IS IMPORTANT.

```typescript
export type HttpMutationOptions<Parameter, Result> = Omit<
  RxMutationOptions<Parameter, NoInfer<Result>>,
  'operation'
> & {
  request: (param: Parameter) => HttpMutationRequest;
  parse?: (response: unknown) => Result;
};
```

Reply:
done